troposphere/static/js/actions/instance/stop.js

```diff
@@ -1,58 +1,88 @@
+import Backbone from "backbone";
 
 import InstanceConstants from "constants/InstanceConstants";
+import Instance from "models/Instance";
 import InstanceState from "models/InstanceState";
-import Utils from "../Utils";
 import InstanceActionRequest from "models/InstanceActionRequest";
 
-export default {
+import Utils from "../Utils";
 
-    stop: function(params) {
-        if (!params.instance)
-            throw new Error("Missing instance");
-
-        var instance = params.instance,
-            instanceState = new InstanceState({
-                status_raw: "active - powering-off",
-                status: "active",
-                activity: "powering-off"
-            }),
-            originalState = instance.get("state"),
-            actionRequest = new InstanceActionRequest({
-                instance: instance
-            });
+/**
+ * Defines the `stop` steps for a single instance
+ */
+function handleOne(instance){
+    // if we've been passed a model via "selected resources" that
+    // is *not* an Instance, then just skip it - _move on_
+    if (!(instance instanceof Instance)) return;
```

Comment:
I don't believe it affects the cardinality, rather it introduces a `[..., undefined, ...]`, which causes other errors. (In this case I understand that we weren't using the result of the `map` making this essentially a `forEach`)

I don't want to badger, because I know that the scope of this PR changed from being a quick release to something we want to invest more time into. But I want to comment on thoughts I feel strongly about (which may just be preaching to the choir).

The context for this code (the fact that it is an instance action) `js/actions/instance/stop.js` means that the action shouldn't be called with a volume, that's just an error upstream. you wouldn't want to fail silently if someone does `launch(volume)`. 

We have the following pattern in quite a few places: 
`function(arg) { if (condition) { bail } ...}`.  Sometimes it make sense, when the function doesn't have to do any work for the argument, i.e. the function is a noOp for that argument. For example: `[].concat([])`. But if condition is `arg is not what I expected at all`, then we're just hiding potential errors.